diff -crB openssh-5.6p1/sftp-common.c openssh-5.6p1-patch/sftp-common.c
*** openssh-5.6p1/sftp-common.c	Sat Jan 16 11:43:50 2010
--- openssh-5.6p1-patch/sftp-common.c	Sun Sep 26 00:11:52 2010
***************
*** 58,63 ****
--- 58,64 ----
  	a->perm = 0;
  	a->atime = 0;
  	a->mtime = 0;
+ 	a->ino = 0;
  }
  
  /* Convert from struct stat to filexfer attribs */
***************
*** 76,81 ****
--- 77,84 ----
  	a->flags |= SSH2_FILEXFER_ATTR_ACMODTIME;
  	a->atime = st->st_atime;
  	a->mtime = st->st_mtime;
+ 	a->flags |= SSH2_FILEXFER_ATTR_EXTENDED;
+ 	a->ino = st->st_ino;
  }
  
  /* Convert from filexfer attribs to struct stat */
***************
*** 96,101 ****
--- 99,106 ----
  		st->st_atime = a->atime;
  		st->st_mtime = a->mtime;
  	}
+ 	if (a->flags & SSH2_FILEXFER_ATTR_EXTENDED)
+ 		st->st_ino = a->ino;
  }
  
  /* Decode attributes in buffer */
***************
*** 152,157 ****
--- 157,167 ----
  		buffer_put_int(b, a->atime);
  		buffer_put_int(b, a->mtime);
  	}
+ 	if (a->flags & SSH2_FILEXFER_ATTR_EXTENDED) {
+ 		buffer_put_int(b, 1);			/* count */
+ 		buffer_put_cstring(b, "ino");	/* type */
+ 		buffer_put_int64(b, a->ino);	/* data */
+ 	}
  }
  
  /* Convert from SSH2_FX_ status to text error message */
diff -crB openssh-5.6p1/sftp-common.h openssh-5.6p1-patch/sftp-common.h
*** openssh-5.6p1/sftp-common.h	Wed Jan 13 20:44:06 2010
--- openssh-5.6p1-patch/sftp-common.h	Sun Sep 26 00:11:52 2010
***************
*** 39,44 ****
--- 39,45 ----
  	u_int32_t	perm;
  	u_int32_t	atime;
  	u_int32_t	mtime;
+ 	u_int64_t	ino;
  };
  
  void	 attrib_clear(Attrib *);
